SweetsFactory/src/app/shared/services/sharing.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class SharingService {
```

Comment:
Because there are situation like: saving data issues, getting data issues; I suggest to implement this service using the promises API